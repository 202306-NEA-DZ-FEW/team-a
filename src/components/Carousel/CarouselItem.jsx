import Link from "next/link";

export default function CarouselItem({
  t,
  name,
  dataKey,
  imageURL,
  description,
  queryParams,
}) {
  return (
    <div
      className='item w-20 h-20 lg:w-[200px] lg:h-1/2 shadow-2xl absolute z-10 lg:top-[50%] top-[80%] lg:-translate-y-[40%] lg:translate-x-[-30%] -translate-x-[170%] bg-blend-overlay bg-black bg-opacity-20'
      style={{ backgroundImage: `url(${imageURL})` }}
    >
      <div className='content absolute z-10 top-[50%] left-[5%] lg:max-w-md text-white -translate-y-[50%] hidden'>
        <h1 className='text-4xl font-bold my-4 name'>{name}</h1>
        <p className='text-lg font-light mb-4 des'>
          {description ? description : ""}
        </p>
        <Link
          className='btn btn-sm bg-white text-black hover:bg-slate-50'
          scroll={false}
          href={{
            pathname: "/products",
            query: { ...queryParams, category: dataKey },
          }}
        >
          {t("productsPage:seeMore")}
        </Link>
      </div>
    </div>
  );
}
