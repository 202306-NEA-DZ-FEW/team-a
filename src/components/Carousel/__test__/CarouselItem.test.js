import image from "public/images/mepi1.png";
import renderer from "react-test-renderer";

import CarouselItem from "../CarouselItem";

it("renders correctly", () => {
  const name = "name";
  const dataKey = "myKey";
  const imageURL = image;
  const description = "my description";
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(
      <CarouselItem
        key={0}
        name={name}
        dataKey={dataKey}
        imageURL={imageURL}
        description={description}
        queryParams={queryParams}
        t={mockT}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
