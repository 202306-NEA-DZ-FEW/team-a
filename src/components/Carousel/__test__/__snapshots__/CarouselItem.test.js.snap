// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="item w-20 h-20 rounded-2xl lg:rounded-3xl lg:w-[220px] lg:h-80 shadow-2xl absolute z-10 lg:top-[50%] top-[80%] lg:-translate-y-[22%] lg:translate-x-[0%] -translate-x-[170%] bg-blend-overlay bg-black bg-opacity-20"
  style={
    Object {
      "backgroundImage": "url([object Object])",
    }
  }
>
  <div
    className="content absolute z-10 top-[50%] left-[5%] lg:max-w-md text-white -translate-y-[50%] hidden"
  >
    <h1
      className="text-4xl font-bold my-4 name"
    >
      name
    </h1>
    <p
      className="text-lg font-light mb-4 des"
    >
      my description
    </p>
    <a
      className="btn btn-sm rounded-xl bg-transparent text-white hover:bg-slate-50 hover:text-black"
      href="#items"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;
